datalad/interface/common_cfg.py

```diff
@@ -197,6 +200,18 @@
                'title': 'Does not execute teardown_package which cleans up temp files and directories created by tests if this flag is set'}),
         'type': EnsureBool(),
     },
+    'datalad.tests.force-fake-home': {
+        'ui': ('yesno', {
+               'title': 'Force running tests on OSX and window with manipulated HOME or USERPROFILE respectively.',
+               'text': "This is potentially dangerous and only applies on OSX and Windows if the git version used by datalad is older than 2.32."
+                       "By default, tests are not executed under these circumstances, since the legacy setup for the test environment can have unintended side-effects:{nl}"
+                       "1. On Windows, subprocesses launched by the tests (incl. indirectly started ones), may end up reading from and writing to wrong locations, as any subprocess will fail to retrieve correct standard locations for configs, caches, etc. This issue is discussed in gh-6160 and gh-6260."
+                       "See also: https://github.com/datalad/datalad/issues/6160 and https://github.com/datalad/datalad/pull/6260 for more details.{nl}"
+                       "2. OSX, most notably the system-level osxkeychain git-credential helper, which will lead to hanging git commands if it fails to find the keychain in its expected location under $HOME/Library/Keychain."
```

Comment:
```suggestion
                       "2. On OSX, the system-level `osxkeychain` git-credential helper will cause hanging Git commands, because it fails to find the keychain under $HOME/Library/Keychain."
```

I never asked, but I assume it is not an option to just copy that keychain?

Reply:
It is. But: I have no idea whether that path is fix or something that we'd need to dynamically find out, possibly opening just another rabbit hole. Plus, that approach would need to be repeated for every such third-party software that we discover over time to have a similar issue. Therefore, that approach felt to me like starting a never-ending story.

Reply:
But on a sample size of 1, testing with a plain copy worked as well.

Reply:
To add one more aspect to this: I don't know whether the copied keychain would be fully functioning. I only tested that such a copy would not lead to a hanging credential helper. If it contains credentials to clone from, say, github, and would actually need to be used during other tests obtaining test repositories or something - I don't know whether there would need to be an additional step of registering/verifying the new location.

Reply:
Before discussing these details, I think it is worth discussing the general approach of removing any isolations. They were introduced to prevent pollution. And now we are saying

- we must not pollute the git config
- we will pollute everything else

it feels a bit schizophrenic to me.

Reply:
Yes, from my point of view it's just about removing the wrong approach of isolation (since it actually breaks stuff + is partially dysfunctional).

I thought about patching the `AppDirs` object from within `setup_package` to make it refer to an application "datalad-test", rather than "datalad" and register the resulting dirs with `_TEMP_PATHS_GENERATED` for clean up.
But that leaves the issue of subprocesses, that are datalad processes, too, and wouldn't have access to the patched one. Like special remotes and credential helper.